transit_odp/publish/tests/test_seasonal_service_views.py

```diff
@@ -147,6 +148,39 @@ def test_delete_last_seasonal_service(self, publish_client):
             == self.NUMBER_SEASONAL_SERVICE_PER_ORGANISATION - 1
         )
 
+    def test_transition_page_when_delete_seasonal_service(self, publish_client):
+        SEASONAL_SERVICE_UPGRADE = 6
+        publish_client.force_login(user=self.user)
+        for i in range(SEASONAL_SERVICE_UPGRADE):
+            SeasonalServiceFactory(licence=self.licence)
+        # up to 3 pages
+        response = publish_client.get(self.url + "?page=3")
+
+        assert response.status_code == 200
+        assert response.context.get("table").paginator.num_pages == 3
+        assert (
+            response.context.get("table").data.model.objects.count()
+            == self.NUMBER_SEASONAL_SERVICE_PER_ORGANISATION + SEASONAL_SERVICE_UPGRADE
+        )
+
+        seasonal_services = SeasonalService.objects.last()
+        delete_data = {"id": seasonal_services.id}
+        delete_url = reverse(
+            "delete-seasonal-service",
+            host=PUBLISH_HOST,
+            kwargs={"pk1": self.org.id},
+        )
+        publish_client.post(
+            delete_url,
+            data=delete_data,
+        )
+        # the 3rd page no longer exists!
+        response = publish_client.get(self.url + "?page=3")
```

Comment:
might be an idea to explicitly test the redirect url to make sure it exists and has seasonal services on it

Reply:
There is no redirect here. The link `self.url + "?page=3"` doesn't exist! and the `follow = True` go straight to 404. 
This pass the test:

`        delete_url = reverse(
            "delete-seasonal-service",
            host=PUBLISH_HOST,
            kwargs={"pk1": self.org.id},
        )
        response = publish_client.post(
            delete_url,
            data=delete_data,
        )
        # redirect to list table
        assert response.status_code == 302
        assert response.url == self.url + "?page=&"
        # the 3rd page no longer exists!
        response = publish_client.get(
            self.url + "?page=3",
            follow=True,
        )
        assert response.status_code == 404